Sources/MusicPlayer/Players/MPRIS.swift

```diff
@@ -7,145 +7,110 @@
-                                          gint64,
-                                          UnsafeMutableRawPointer?) -> Void
-                = { player, position, data in
-                    data?.unretainedCast(to: MPRIS.self).updatePlayerState()
-                }
-            
-            let onMetadataChanged: @convention(c) (UnsafeMutablePointer<PlayerctlPlayer>?,
-                                                   OpaquePointer? /* GVariant* */,
-                                                   UnsafeMutableRawPointer?) -> Void
-                = { player, metadata, data in
-                    data?.unretainedCast(to: MPRIS.self).updatePlayerState()
-                }
-            
-            let pself = Unmanaged.passUnretained(self).toOpaque()
-            signals.append(
-                g_signal_connect_data(player, "playback-status",
-                                      unsafeBitCast(onPlayStatusChanged, to: GCallback?.self), pself, nil, G_CONNECT_AFTER)
-            )
-            signals.append(
-                g_signal_connect_data(player, "seeked", unsafeBitCast(onSeeked, to: GCallback?.self), pself, nil, G_CONNECT_AFTER)
-            )
-            signals.append(
-                g_signal_connect_data(player, "metadata", unsafeBitCast(onMetadataChanged, to: GCallback?.self), pself, nil, G_CONNECT_AFTER)
-            )
+            let updatePlayerState: () -> Void = { [weak self] in self?.updatePlayerState() }
+            disposables.append(try player.player.playbackStatus.observe(updatePlayerState))
+            disposables.append(try player.player.metadata.observe(updatePlayerState))
+            disposables.append(try player.player.seeked { _ in updatePlayerState() })
             updatePlayerState()
         }
-        
+
         deinit {
-            for var signal in signals {
-                g_clear_signal_handler(&signal, player)
-            }
-            g_object_unref(player)
+            disposables.forEach { try? $0() }
         }
     }
 }
 
 extension MusicPlayers.MPRIS {
-    
-    public class var names: [String] {
-        let playerNames = playerctl_list_players(nil)
-        var result: [String] = []
-        var cur = playerNames
-        while cur != nil {
-            let playerName = cur!.pointee.data.assumingMemoryBound(to: PlayerctlPlayerName.self)
-            result.append(String(cString: playerName.pointee.name))
-            playerctl_player_name_free(playerName)
-            cur = cur!.pointee.next
-        }
-        g_list_free(playerNames)
-        return result
+    /// Initializes a new MPRIS music player.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the media player.
+    ///   - instance: The instance name of the media player, if any.
+    ///   - queue: The dispatch queue for the D-Bus method calls.
+    public convenience init(
+        name: String, instance: String? = nil, queue: DispatchQueue? = nil
```

Comment:
I think it's better to make it as simple as possible, treat queue, connection, and timeout as implementation details and don't expose them. We can always expose them later if someone needs it.